drivers/tty/serial/serial_core.c

```diff
@@ -3457,7 +3464,7 @@ int serial_core_register_port(struct uart_driver *drv, struct uart_port *port)
 err_unregister_ctrl_dev:
 	serial_base_ctrl_device_remove(new_ctrl_dev);
 
-	if (strncmp(to_platform_device(port->dev)->name, "anon_port", 9) == 0)
+	if (is_parent_null)
```

Comment:
nitpick: Seeing that if the parent is not NULL anymore after `port->dev = &pdev->dev;` above, naming this `is_parent_anon_port` or something like that might be better.

Reply:
Fixed.